core/dialog.ts

```diff
@@ -112,3 +112,12 @@ export function setPrompt(
             AnyDuringMigration) {
   promptImplementation = promptFunction;
 }
+
+function confirmInternal(
+    message: string, callback: (p1: boolean) => AnyDuringMigration) {
+  confirmImplementation(message, callback);
+}
+
+export const TEST_ONLY = {
+  confirmInternal,
+}
```

Comment:
This seems like avoidable redirection.  Do we need to ensure that we can correctly stub (and unstub) `confirm` even in the face of calls to `setConfirm`?

If not, why not just turn `confirmImplementation` into a property on `TEST_ONLY` and omit `confirmInternal` entirely?

Reply:
Yeah I thought about just using the `confirmImplementation` here, but I wanted to make it clear that that was an actual feature for dev swapping in their own modal code, whereas we just need something to get tests to run.

Right now I prefer having the extra layer of indirection to indicate that we dealing with a non-ideal situation. I don't want to try to make good final decisions atm.

Reply:
That seems reasonable.